MathLib/LinAlg/UnifiedMatrixSetters.cpp

```diff
@@ -89,9 +89,27 @@ void setVector(Eigen::VectorXd& v, MatrixVectorTraits<Eigen::VectorXd>::Index co
 namespace MathLib
 {
 
-double norm(PETScVector const& x)
+double norm(PETScVector const& x, MathLib::VecNormType nmtype = MathLib::VecNormType::NORM2)
```

Comment:
Suggest to remove this method from this file (maybe in a later PR).
I am against default arguments. They only make the code more error-prone.
I have some code regarding norms in a private branch. I'll make a PR in the next few weeks.